core/app/models/spree/credit_card.rb

```diff
@@ -29,14 +29,22 @@ class CreditCard < Spree::Base
     # needed for some of the ActiveMerchant gateways (eg. SagePay)
     alias_attribute :brand, :cc_type
 
+    # Taken from ActiveMerchant
+    # https://github.com/activemerchant/active_merchant/blob/2f2acd4696e8de76057b5ed670b9aa022abc1187/lib/active_merchant/billing/credit_card_methods.rb#L5
     CARD_TYPES = {
-      visa: /^4[0-9]{12}(?:[0-9]{3})?$/,
-      master: /(^5[1-5][0-9]{14}$)|(^6759[0-9]{2}([0-9]{10})$)|(^6759[0-9]{2}([0-9]{12})$)|(^6759[0-9]{2}([0-9]{13})$)/,
-      diners_club: /^3(?:0[0-5]|[68][0-9])[0-9]{11}$/,
-      american_express: /^3[47][0-9]{13}$/,
-      discover: /^6(?:011|5[0-9]{2})[0-9]{12}$/,
-      jcb: /^(?:2131|1800|35\d{3})\d{11}$/
-    }
+      'visa'               => /^4\d{12}(\d{3})?(\d{3})?$/,
```

Comment:
Why not take the Regex from ActiveMerchant instead of copy it over. This way we have to always have to have an eye on this.